applications/CompressiblePotentialFlowApplication/python_scripts/potential_flow_response.py

```diff
@@ -0,0 +1,322 @@
+            raise RuntimeError("_SynchronizeAdjointFromPrimal: Model parts have a different number of nodes!")
+
+        # TODO this should happen automatically
+        for primal_node, adjoint_node in zip(self.primal_model_part.Nodes, self.adjoint_model_part.Nodes):
+            adjoint_node.X0 = primal_node.X0
+            adjoint_node.Y0 = primal_node.Y0
+            adjoint_node.Z0 = primal_node.Z0
+            adjoint_node.X = primal_node.X
+            adjoint_node.Y = primal_node.Y
+            adjoint_node.Z = primal_node.Z
+
+        # Put primal solution on adjoint model
+        if self.primal_data_transfer_with_python:
+            Logger.PrintInfo(self._GetLabel(), "Transfer primal state to adjoint model part.")
+            variable_utils = KratosMultiphysics.VariableUtils()
+            for variable in self.primal_state_variables:
+                variable_utils.CopyModelPartNodalVar(variable, self.primal_model_part, self.adjoint_model_part, 0)
+
+
+    def _GetAdjointParameters(self):
+        adjoint_settings = self.response_settings["adjoint_settings"].GetString()
+        with open(self.response_settings["adjoint_settings"].GetString(),'r') as parameter_file:
+            adjoint_parameters = Parameters( parameter_file.read() )
+
+        return adjoint_parameters
+
+    def _GetLabel(self):
+        type_labels = {
+            "adjoint_lift_potential_jump" : "LiftPotentialJump"
+        }
+        response_type = self.response_settings["response_type"].GetString()
+        return "Adjoint" + type_labels[response_type]  +"Response"
+
+    def _CheckParameters(self, parameters):
+        if not parameters["solver_settings"].Has("reform_dofs_at_each_step") or not parameters["solver_settings"]["reform_dofs_at_each_step"].GetBool():
+            if not parameters["solver_settings"].Has("reform_dofs_at_each_step"):
+                parameters["solver_settings"].AddEmptyValue("reform_dofs_at_each_step")
+            parameters["solver_settings"]["reform_dofs_at_each_step"].SetBool(True)
+            wrn_msg = 'This solver requires the setting reform the dofs at each step in optimization.'
+            wrn_msg += 'The solver setting has been set to True'
+            Logger.PrintWarning(self._GetLabel(), wrn_msg)
+        for subproc_keys, subproc_values in parameters["processes"].items():
+            for process  in subproc_values:
+                if "wake" in process["python_module"].GetString():
+                    if not process["Parameters"].Has("compute_wake_at_each_step") or not process["Parameters"]["compute_wake_at_each_step"].GetBool():
+                        if not process["Parameters"].Has("compute_wake_at_each_step"):
+                            process["Parameters"].AddEmptyValue("compute_wake_at_each_step")
+                    process["Parameters"]["compute_wake_at_each_step"].SetBool(True)
+        return parameters
+
+
+class AngleOfAttackResponseFunction(ResponseFunctionInterface):
```

Comment:
I would move the 3 geometric responses to a separate file, ev. `2d_airfoil_responses.py`

Reply:
Then i would even consider moving them to the shape-opt application since this could also be used by @sunethwarna  right?

Reply:
> I would move the 3 geometric responses to a separate file, ev. `2d_airfoil_responses.py`

Ok, it makes sense!


> Then i would even consider moving them to the shape-opt application since this could also be used by @sunethwarna right?

Yes! What do you think @sunethwarna ? Were you already using some sort of geometrical constraints?

Reply:
I have implemented followings:

https://github.com/KratosMultiphysics/Kratos/blob/rans/feature/adjoint/vortex_shedding_frequency_sensitivities/applications/ShapeOptimizationApplication/python_scripts/response_functions/geometric_centroid_deviation.py

https://github.com/KratosMultiphysics/Kratos/blob/rans/feature/adjoint/vortex_shedding_frequency_sensitivities/applications/ShapeOptimizationApplication/python_scripts/response_functions/total_volume.py

At the moment they are in my branches (yet to be merged). If you think that they are also useful for your cases, then I can move them out from my branch and create a PR to put them in master. What do you suggest?

Reply:
Yes those could be useful as well! I think it is worth moving them to the ShapeApp.

What is the main purpose of constraining the centroid? Just to limit translations of the geometry?

Reply:
> What is the main purpose of constraining the centroid? Just to limit translations of the geometry?

Yes.

> Yes those could be useful as well! I think it is worth moving them to the ShapeApp.

I'll add them as well :)

Reply:
@armingeiser should I put the "airfoil_responses" into the shape app in this PR? Or do you think it is better to do it in a second PR?

Reply:
You can put them there in this PR IMO